src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -404,7 +404,7 @@ export class EditorGroupModel extends Disposable {
 	}
 
 	private registerEditorListeners(editor: EditorInput): void {
-		const listeners = new DisposableStore();
+		const listeners = this._register(new DisposableStore());
```

Comment:
With this code, we now store a `DisposableStore` for each editor that gets opened and closed, even when the disposable has been disposed. So we grow an array for every editor until you close the window. That is the reason I think I was not registering the disposable globally.